Loop/Managers/DeviceDataManager.swift

```diff
@@ -1357,84 +1366,74 @@ extension Notification.Name {
 }
 
 // MARK: - Remote Notification Handling
-extension DeviceDataManager {
+
+extension DeviceDataManager: RemoteCommandHandler {
     
     func handleRemoteNotification(_ notification: [String: AnyObject]) {
         Task {
-            let backgroundTask = await beginBackgroundTask(name: "Remote Data Upload")
-            await handleRemoteNotification(notification)
+            log.default("Remote Notification: Handling notification %{public}@", notification)
+            
+            guard FeatureFlags.remoteCommandsEnabled else {
+                log.error("Remote Notification: Remote Commands not enabled.")
+                return
+            }
+            
+            let backgroundTask = await beginBackgroundTask(name: "Handle Remote Notification")
+            do {
+                try await remoteDataServicesManager.handleRemoteNotification(notification)
+            } catch {
+                log.error("Remote Notification: Error: %{public}@", String(describing: error))
+            }
+            
             await endBackgroundTask(backgroundTask)
+            log.default("Remote Notification: Finished handling")
         }
     }
     
-    func handleRemoteNotification(_ notification: [String: AnyObject]) async {
-        
-        defer {
-            log.default("Remote Notification: Finished handling")
-        }
-        
-        guard FeatureFlags.remoteCommandsEnabled else {
-            log.error("Remote Notification: Remote Commands not enabled.")
-            return
-        }
-        
-        let command: RemoteCommand
-        do {
-            command = try await remoteDataServicesManager.commandFromPushNotification(notification)
-        } catch {
-            log.error("Remote Notification: Parse Error: %{public}@", String(describing: error))
-            return
+    func handlePendingRemoteCommands() {
+        Task {
+            guard FeatureFlags.remoteCommandsEnabled else {
+                return
+            }
+            
+            let backgroundTask = await beginBackgroundTask(name: "Handle Pending Remote Commands")
+            await remoteDataServicesManager.handlePendingRemoteCommands()
+            await endBackgroundTask(backgroundTask)
         }
-        
-        await handleRemoteCommand(command)
     }
     
-    func handleRemoteCommand(_ command: RemoteCommand) async {
+    //Remote Overrides
+    
```

Comment:
With the removal of the `OverrideAction` type, this validation logic needed moved into here.